pkg/test/discovery_client.go

```diff
@@ -80,3 +81,11 @@ func (c *DiscoveryClient) WithAPIResource(resource *APIResource) *DiscoveryClien
 
 	return c
 }
+
+func (c *DiscoveryClient) GroupsAndMaybeResources() (*metav1.APIGroupList, map[schema.GroupVersion]*metav1.APIResourceList, map[schema.GroupVersion]error, error) {
+	apiGroupList, err := c.ServerGroups()
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	return apiGroupList, nil, nil, nil
+}
```

Comment:
This function isn't referenced anywhere?

Reply:
This function is required for `DiscoveryClient` as a mock implementation for interface `discovery.AggregatedDiscoveryInterface`.

It's called inside the `client-go/discovery` code

Reply:
gotcha. makes sense.